src/network/networkd-dhcp-static-lease.c

```diff
@@ -0,0 +1,127 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "hashmap.h"
+#include "memory-util.h"
+#include "networkd-dhcp-static-lease.h"
+#include "networkd-util.h"
+#include "strv.h"
+#include "utf8.h"
+
+DHCPStaticLease *dhcp_static_lease_free(DHCPStaticLease *static_lease) {
+        if (!static_lease)
+                return NULL;
+
+        if (static_lease->network) {
+                LIST_REMOVE(dhcp_static_leases, static_lease->network->dhcp_static_leases, static_lease);
+                assert(static_lease->network->n_dhcp_static_leases > 0);
+                static_lease->network->n_dhcp_static_leases--;
+
+                if (static_lease->section)
+                        hashmap_remove(static_lease->network->dhcp_static_leases_by_section, static_lease->section);
+        }
+
+        network_config_section_free(static_lease->section);
+        return mfree(static_lease);
+}
+
+int dhcp_static_lease_new(DHCPStaticLease **ret) {
+        DHCPStaticLease *p = NULL;
+        struct in_addr address = (struct in_addr) {
+                .s_addr = 0,
+        };
+
+        assert_return(ret, -EINVAL);
+
+        p = new(DHCPStaticLease, 1);
+        if (!p)
+                return -ENOMEM;
+
+        /* Static lease address can be omitted in the config,
+         * removing any static lease with the specified mac address. */
+        *p = (DHCPStaticLease) {
+                .address = address,
```

Comment:
no need to initialize this? it's zero anyway... just use new0() instead of new() if all fields are supposed to be zero anyway